code/game/turfs/space/space.dm

```diff
@@ -15,6 +15,13 @@
 /turf/space/New()
 	if(!istype(src, /turf/space/transit))
 		icon_state = "[((x + y) ^ ~(x * y) + z) % 25]"
+	update_starlight()
+
+/turf/space/proc/update_starlight()
+	for(var/turf/T in range(src,1))
+		if(istype(T,/turf/simulated))
+			src.SetLuminosity(2)
+			return
 

```

Comment:
Well, it's called when deconstructing, because all it does is make a new space turf, which calls update_starlight(). Constructing is the issue, also deleting turfs